examples/parallel_profiling.py

```diff
@@ -0,0 +1,144 @@
+"""
+================================================
+Memory and CPU consumption using joblib.Parallel
+================================================
+
+This example illustrates some important aspects to keep in mind when using
+:class:`joblib.Parallel`. We will focus on different patterns resulting to the
+same outcomes but which trigger different internal behaviors. Those differences
+will impact the memory consumption and computation time.
+
+To illustrate those differences, the moving average from a large array will be
+computed. Therefore, the average can be computed in parallel at different
+location of the signal.
+
+"""
+from __future__ import division
+
+import os
+import time
+import gc
+import numpy as np
+from memory_profiler import memory_usage
```

Comment:
I would not do all the imports at the top, but only in the cell in which they are used. It reduces cognitive load of the reader (though it's bad development practice)